tests/integration/sharding_tests/helpers.py

```diff
@@ -15,6 +22,56 @@
 SHARD_REL_NAME = "sharding"
 CLUSTER_COMPONENTS = [SHARD_ONE_APP_NAME, SHARD_TWO_APP_NAME, CONFIG_SERVER_APP_NAME]
 TIMEOUT = 15 * 60
+MONGOS_PORT = 27018
+MONGOD_PORT = 27017
+
+
+async def generate_mongodb_client(
```

Comment:
I'm 50/50 on recommending this - so its up to you

Should this function make use of our [already existing mongodb_client creator](https://github.com/canonical/mongodb-k8s-operator/blob/c0526a07a300813edafae50e52fe41eb8dc9f303/tests/integration/ha_tests/helpers.py#L377)

obv the original would need to be updated too

Reply:
I wondered the same thing, I can reuse the already existing mongodb-client-creator but it needs quite a few updates.
Let me think about it a few minutes and I'll tell you.

Reply:
Ok I'm doing it :)